src/hid/hid-device.cpp

```diff
@@ -161,10 +161,30 @@ namespace librealsense
                         {
                             if(!_running)
                                 return;
-                            if(r->get_actual_length() == sizeof(REALSENSE_HID_REPORT))
+                            
+                            if( r->get_actual_length() == _realsense_hid_report_actual_size )
                             {
                                 REALSENSE_HID_REPORT report;
-                                memcpy(&report, r->get_buffer().data(), r->get_actual_length());
+                                if( _realsense_hid_report_actual_size != sizeof( REALSENSE_HID_REPORT ) )
+                                {
+                                    //for FX version < 5.16 the actual struct is 32 bit, so we can not use memcpy for the whole struct
```

Comment:
FX --> FW

Reply:
Why do we need to lines that say 
for FX version < 5.16
...
for FX version < 5.16
....?